include/dd/Package.hpp

```diff
@@ -1952,62 +1958,82 @@ template <class Config> class Package {
 
   bool isCloseToIdentityRecursive(const mEdge& m,
                                   std::unordered_set<decltype(m.p)>& visited,
-                                  const dd::fp tol) {
+                                  const dd::fp tol,
+                                  const std::vector<bool>& garbage,
+                                  const bool checkCloseToOne) {
     // immediately return if this node has already been visited
     if (visited.find(m.p) != visited.end()) {
       return true;
     }
 
-    // immediately return of this node is identical to the identity
+    // immediately return if this node is identical to the identity
     if (m.isTerminal() || m.p->isIdentity()) {
       return true;
     }
 
+    const auto n = m.p->v;
+
+    if (garbage.size() > n && garbage[n]) {
+      return isCloseToIdentityRecursive(m.p->e[0U], visited, tol, garbage,
+                                        checkCloseToOne) &&
+             isCloseToIdentityRecursive(m.p->e[1U], visited, tol, garbage,
+                                        checkCloseToOne) &&
+             isCloseToIdentityRecursive(m.p->e[2U], visited, tol, garbage,
+                                        checkCloseToOne) &&
+             isCloseToIdentityRecursive(m.p->e[3U], visited, tol, garbage,
+                                        checkCloseToOne);
+      ;
+    }
+
     // check whether any of the middle successors is non-zero, i.e., m = [ x 0 0
     // y ]
     const auto mag1 = dd::ComplexNumbers::mag2(m.p->e[1U].w);
     const auto mag2 = dd::ComplexNumbers::mag2(m.p->e[2U].w);
     if (mag1 > tol || mag2 > tol) {
-      visited.insert(m.p);
-      return false;
-    }
-
-    // check whether  m = [ ~1 0 0 y ]
-    const auto mag0 = dd::ComplexNumbers::mag2(m.p->e[0U].w);
-    if (std::abs(mag0 - 1.0) > tol) {
-      visited.insert(m.p);
-      return false;
-    }
-    const auto arg0 = dd::ComplexNumbers::arg(m.p->e[0U].w);
-    if (std::abs(arg0) > tol) {
-      visited.insert(m.p);
+      visited.insert(m.p); // is that not superfluous?
```

Comment:
I see your point. It could be worth removing this to keep the visited set smaller.
Such a node that is close to zero wouldn't be visited anyway, I guess.
Does anything seem to break if you remove the statement?

Reply:
It seems to work when I remove it. Anyway after this insert, the function returns immediately, and the visited set goes out of scope, so I don't think this can cause probems.